app/addons/dataimporter/components.react.jsx

```diff
@@ -0,0 +1,793 @@
+
+    rows: function () {
+      var data = this.props.data;
+
+      if (this.props.isBigFile) {
+        data = this.props.getSmallPreviewOfData;
+      }
+
+      return (
+        data.map(function (dataObj, i) {
+          var obj = this.props.getHeaderConfig ? dataObj :
+            this.objectify(dataObj);
+          if (i < 500) {
+            return (
+              <Components.SimpleDoc
+                id={"<UUID>_" + i}
+                content={JSON.stringify(obj, null, ' ')}
+                key={i} />
+            );
+          }
+        }.bind(this))
+      );
+    },
+
+    render: function () {
+      if (this.props.getPreviewView === "json") {
+        return (
+          <div id="doc-list" className="json-view">
+            {this.rows()}
+          </div>
+        );
+      } else {
+        return null;
+      }
+    }
+  });
+
+  var Footer = React.createClass({
+    getInitialState: function () {
+      return {
+        targetDB: this.props.getAllDBs[0],
+        selectExistingDB: true
+      };
+    },
+
+    startOverButton: function () {
+      return (
+        <a className="start-import-over-link footer-button"
+          onClick={this.startover}>
+          <span className="fonticon icon-repeat"></span>
+            Start Over
+        </a>
+      );
+    },
+
+    startover: function () {
+      Actions.dataImporterInit(true);
+    },
+
+    getExistingDBList: function () {
+      var allDBs = this.props.getAllDBs,
+          options = [],
+          setTargetDB = this.setTargetDB;
+
+      _.each(allDBs, function (dbName, i) {
+        options.push({
+          name: dbName,
+          onClick: function () { setTargetDB(dbName); }
+        });
+      });
+
+      return options;
+    },
+
+    setTargetDB: function (dbName) {
+      this.setState({ targetDB: dbName });
+    },
+
+    newOrExistingToggle : function () {
+      var config = {
+        title: 'Load into',
+        leftLabel : 'Existing database',
+        rightLabel : 'New database',
+        defaultLeft: true,
+        leftClick: function () {
+          this.setState({
+            selectExistingDB: true,
+            targetDB: this.props.getAllDBs[0]
+          });
+        }.bind(this),
+        rightClick: function () {
+          this.setState({ selectExistingDB: false });
+        }.bind(this),
+        enclosingID: 'choose-database-to-load-data-into'
+      };
+
+      return <Components.ToggleState toggleConfig={config} />;
+    },
+
+    chooseDatabaseFromDropdown : function () {
+      var selected = this.state.targetDB;
+
+      var setup = {
+        title: 'Choose a database',
+        id: 'data-importer-choose-db',
+        selected: selected,
+        selectOptions: this.getExistingDBList()
+      };
+      return <Components.SmallDropdown dropdownSetup={setup} />;
+    },
+
+    createNewDB: function () {
+      return (
+        <div id="data-import-name-new-target-database">
+          <div id="title">Name New Database</div>
```

Comment:
My CSS friend would shoot you for the lack of semantics here (as he's shot me). :) Titles should be <h1>, <h2>, etc. Can be a bit irritating if you have to reset the CSS on them a bit, but it definitely makes for more readable markup.